### ISSUE TITLE
Severe performance problems on alt path (mirrored world)

### ISSUE THREAD

**Issue by Tvangeste** [TazTxUK/MinimapAPI#147]
This happens almost every time, and esp. if I exit/save and return back a couple of times.
Eventually, the game is loading for many minutes and very stuttery.

The minimapi/save1.dat quickly become HUGE. For example, my current one is 420KB when I spent some time in Downpour || in mirrored world. When looking into the file, I see the LevelData structure, consisting of two arrays, the first one is of length 1632 (!!!) and rapidly growing :)

The second array is of size 34, which seems to be normal.

Basically, with this problem it is close to impossible for me to complete the run. I had to disable the MinimapAPI mod when I am in Downpour II, which is a hassle.

Thank you!

P.S. Also, navigating the mirrored world's map is weird, the left switch with the right and the map is mirrored, makes it awkward to teleport :)

**Comment by Zamiell** [comment]
does the problem happen with only minimap api enabled and all other mods disabled?

**Comment by Tvangeste** [comment]
The problem happens in both cases, when I disable all mods except the minimapAPI, and when I enable all my typical mods.

I was removing mods one by one, and the only one that causes the problems in this scenario (mirror world with multiple pause-exits).

And as soon as I leave the level to the next one, the problem is gone altogether.

**Comment by Lunastela** [comment]
Would like to bring attention to this. It seems to be related to any dimensions that aren't the main dimension, Unsure if it is a direct result of rewinding/using Glowing Hourglass in Mirror World / Mines Puzzle, but that seems to be the easiest way to trigger it. It causes Save File data to increase until it begins to lag the game.
